src/core/config/tmf.h

```diff
@@ -42,8 +42,12 @@
  *
  */
 #ifndef OPENTHREAD_CONFIG_TMF_ADDRESS_CACHE_ENTRIES
+#ifdef OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE
```

Comment:
Since we are using the `OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE` directly here, suggest adding this to the top of `tmf.h` (to make use it if always defined).
```
#include "config/border_router.h"
```